### DISCUSSION TITLE
[Bug]:  No apply config to my.cnf

### DISCUSSION THREAD

**Discussion by jAKErCZ** [beyondcode/herd-community#1054] [General]
### Platform

macOS

### Operating system version

macOS Sonoma 14.6.1

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.11.0

### PHP Version

PHP 8.3

### Bug description

hello
I have a problem with mysql configuration in my.cnf where I try to add:

`[mysqld]
sql_mode="STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"
datadir="/Users/korcak/Library/Application Support/Herd/config/services/8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47/"
basedir="/Users/Shared/Herd/services/mysql/8.0.36"
socket="/tmp/mysql-8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47.sock"
log-error="/Users/korcak/Library/Application Support/Herd/config/services/8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47//error.log"
plugin-dir="/Users/Shared/Herd/services/mysql/8.0.36/lib/plugin"
disable_log_bin
`

When I use sql_mode I cannot start the mysql service.

Error mysql:

`2024-10-01T12:45:01.6NZ mysqld_safe Logging to '/Users/korcak/Library/Application Support/Herd/config/services/8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47/Dominik--MacBook-Pro.local.err'.
2024-10-01T12:45:01.6NZ mysqld_safe Starting mysqld daemon with databases from /Users/korcak/Library/Application Support/Herd/config/services/8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47
2024-10-01T12:45:01.6NZ mysqld_safe mysqld from pid file /Users/korcak/Library/Application Support/Herd/config/services/8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47/Dominik--MacBook-Pro.local.pid ended`

What am I doing wrong?

thx

**Comment by mpociot** [comment]
What error is in your log file?

`/Users/korcak/Library/Application Support/Herd/config/services/8A5BCDE3-D5BE-42AB-AC0B-940E6E561A47/Dominik--MacBook-Pro.local.err`

You should remove `NO_AUTO_CREATE_USER` as this was removed in MySQL 8+, likely causing this error.
